src/lib/Zikula/Core/Theme/Engine.php

```diff
@@ -139,20 +147,62 @@ public function wrapResponseInTheme(Response $response)
     }
 
     /**
-     * Wrap a block in the theme's block template.
-     * @api Core-2.0
-     * @todo consider changing block to a Response
-     * @param array $block
-     * @return bool|string (false if theme is not twigBased)
+     * BC method to wrap a block in the theme's block template if theme is twig-based.
+     * @deprecated
+     * @param array $blockInfo
+     * @return string
      */
-    public function wrapBlockInTheme(array $block)
+    public function wrapBcBlockInTheme(array $blockInfo)
     {
         // @todo remove twigBased check in 2.0
         if (!isset($this->activeThemeBundle) || !$this->activeThemeBundle->isTwigBased()) {
             return false;
         }
+        $position = !empty($blockInfo['position']) ? $blockInfo['position'] : 'none';
+        $content = $this->activeThemeBundle->generateThemedBlockContent($this->getRealm(), $position, $blockInfo['content'], $blockInfo['title']);
+
+        return $content;
+    }
+
+    /**
+     * Wrap the block content in the theme block template and wrap that with a unique div.
+     * @api Core-2.0
+     * @param string $content
+     * @param string $title
+     * @param string $blockType
+     * @param integer $bid
+     * @param string $positionName
+     * @param bool $legacy @deprecated param
+     * @return Response
+     */
+    public function wrapBlockContentInTheme($content, $title, $blockType, $bid, $positionName, $legacy)
+    {
+        if (!$legacy) {
+            // legacy blocks are already themed at this point. @todo at Core-2.0 remove $legacy param and this check.
+            $content = $this->activeThemeBundle->generateThemedBlockContent($this->getRealm(), $positionName, $content, $title);
+        }
+
+        // always wrap the block (in the previous versions this was configurable, but no longer) @todo remove comment
+        return $this->wrapBlockContentWithUniqueDiv($content, $positionName, $blockType, $bid);
+    }
 
-        return $this->activeThemeBundle->generateThemedBlock($this->getRealm(), $block);
+    /**
+     * Enclose themed block content in a unique div which is useful in applying styling.
+     *
+     * @param string $content
+     * @param string $positionName
+     * @param string $blockType
+     * @param integer $bid
+     * @return string
+     */
+    private function wrapBlockContentWithUniqueDiv($content, $positionName, $blockType, $bid)
+    {
+        return '<div class="z-block '
```

Comment:
I have already mentioned about this to @craigh This is another remnant of "pre bootstrap age". We do not need to add any html/css at this stage, this can be added in block template. This makes difficult to arrange blocks in block position using bootstrap grid.